api/services/leases/v1/leases.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+
+package containerd.services.leases.v1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/containerd/containerd/api/services/leases/v1;leases";
+
+// Leases service manages resources leases within the metadata store.
+service Leases {
+	// Create creates a new lease for managing changes to metadata. A lease
+	// can be used to protect objects from being removed.
+	rpc Create(CreateRequest) returns (CreateResponse);
+
+	// Delete deletes the lease and makes any unreferenced objects created
+	// during the lease eligible for garbage collection if not referenced
+	// or retained by other resources during the lease.
+	rpc Delete(DeleteRequest) returns (google.protobuf.Empty);
+
+	// ListTransactions lists all active leases, returning the full list of
+	// leases and optionally including the referenced resources.
+	rpc List(ListRequest) returns (ListResponse);
+}
+
+// Snapshot is a snapshot resource reference.
+message Snapshot {
+	string snapshotter = 1;
+	string key = 2;
+}
+
+// Lease is an object which retains resources while it exists.
+message Lease {
+	string id = 1;
+
+	google.protobuf.Timestamp created_at = 2 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+
+	map<string, string> labels = 3;
+
+	repeated Snapshot snapshots = 4;
```

Comment:
Why are snapshots special?

Reply:
I was trying to avoid exposing another microformat for specifying resources. Today in the API we always have snapshotter and snapshot key broken out. We could combine it into a string, but then the client might need to parse it. We combine them into strings a few times in the backend, but do not persist them like that. Do you have an opinion or better suggestion for what to do here?

Reply:
What if we had an object reference:

```
message ObjectRef {
  string plugin_type = 1;
  string plugin_id = 2; // ie btrfs, local, etc.
  string type = 3; // ie snapshot, blob
  string key = 4; // identifier, digest, etc.
}
```

Reply:
I think a reference object makes sense. I am going to remove both of these fields for now and will add them with a proper reference type when I add support in ctr for managing leases